src/test/java/org/openstreetmap/atlas/utilities/configuration/MergedConfigurationTest.java

```diff
@@ -60,101 +88,209 @@ public void testDotFormat() throws IOException
     }
 
     @Test
-    public void testLayered() throws IOException
+    public void testLayeredJson() throws IOException
     {
         final ClassLoader loader = Thread.currentThread().getContextClassLoader();
 
-        try (InputStream base = loader.getResourceAsStream(BASE_CONFIGURATION);
-                InputStream override = loader.getResourceAsStream(OVERRIDE_CONFIGURATION))
+        try (InputStream base = loader.getResourceAsStream(BASE_CONFIGURATION_JSON);
+                InputStream override = loader.getResourceAsStream(OVERRIDE_CONFIGURATION_JSON))
         {
-            final Configuration configuration = new MergedConfiguration(
-                    new InputStreamResource(base), new InputStreamResource(override));
+            testLayered(base, override);
+        }
+    }
+
+    @Test
+    public void testLayeredYaml() throws IOException
+    {
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+
+        try (InputStream base = loader.getResourceAsStream(BASE_CONFIGURATION_YAML);
+                InputStream override = loader.getResourceAsStream(OVERRIDE_CONFIGURATION_YAML))
+        {
+            testLayered(base, override);
+        }
+    }
+
+    @Test
+    public void testLayeredYamlOnJson() throws IOException
+    {
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+
+        try (InputStream base = loader.getResourceAsStream(BASE_CONFIGURATION_JSON);
+                InputStream override = loader.getResourceAsStream(OVERRIDE_CONFIGURATION_YAML))
+        {
+            testLayered(base, override);
+        }
+    }
+
+    @Test
+    public void testLayeredJsonOnYaml() throws IOException
+    {
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+
+        try (InputStream base = loader.getResourceAsStream(BASE_CONFIGURATION_YAML);
+                InputStream override = loader.getResourceAsStream(OVERRIDE_CONFIGURATION_JSON))
+        {
+            testLayered(base, override);
+        }
+    }
+
+    private void testLayered(final InputStream base, final InputStream override)
+    {
+
+        final Configuration configuration = new MergedConfiguration(new InputStreamResource(base),
```

Comment:
If that base input stream will be read more than once, it might cause issues here. It might be better to pass around a supplier that re-creates a new stream at every call, and then pass that supplier to the InputStreamResource constructor.